### PR TITLE
Allow principal search for display name OR email

### PR DESCRIPTION
**Pull request by ChristophWurst**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -274,11 +287,19 @@ export default class DavClient {
...
-	async principalPropertySearch(props, match) {
+	async principalPropertySearch(props, match, test) {
 		const [skeleton, propSearch] = XMLUtility.getRootSkeleton(
-			[NS.DAV, 'principal-property-search'], [NS.DAV, 'property-search']);
+			[NS.DAV, 'principal-property-search'],
+			[NS.DAV, 'property-search']
+		);
+		if (test) {
+			skeleton.attributes = [
+				['test', test]
```
Comment (rullzer) [inline src/index.js:300]: what?

Comment (ChristophWurst) [reply src/index.js:300]: see `\Sabre\DAVACL\AbstractPrincipalCollection::searchPrincipals`. `test` is the argument that tells how the props should be combined, like 'allof' vs 'anyof'

Comment (ChristophWurst) [reply src/index.js:300]: `<x0:principal-property-search xmlns:x0="DAV:" test="anyof">…`

Comment (rullzer) [reply src/index.js:300]: ok.... I was sure this was just a leftover...

**Review by rullzer — APPROVED** [review]

**Review by tcitworld — APPROVED** [review]

**Pull request merged**